lms/util/authorize_lms.py

```diff
@@ -112,18 +111,16 @@ def wrapper(request, *args, **kwargs):
 
         lti_params = lti_params_for(request)
 
-        application_instance = find_by_oauth_consumer_key(
-            request.db, lti_params["oauth_consumer_key"]
-        )
-
-        if application_instance is None:
+        ai_getter = request.find_service(name="ai_getter")
+        consumer_key = lti_params["oauth_consumer_key"]
+        try:
+            client_id = ai_getter.developer_key(consumer_key)
+            client_secret = ai_getter.developer_secret(consumer_key)
+            lms_url = ai_getter.lms_url(consumer_key)
```

Comment:
This is an example of a place which is accessing multiple properties of the app instance and I'm not clear here whether that is going to end up making multiple DB queries when only one should be needed.

Reply:
Yep. I believe this will now make 3 queries